### PR TITLE
Add cpu quotas for AMA

### PR DESCRIPTION
**Pull request by hestolz**
set based on benchmarks and real-world guest agent telemetry

### CODE REVIEW

File: AzureMonitorAgent/HandlerManifest.json [hunk 1]
```diff
@@ -17,21 +17,26 @@
       "services": [
         {
           "name": "azuremonitoragent"
+          "cpuQuotaPercentage": 250
         },
         {
           "name": "azuremonitoragentmgr"
```
Comment (hestolz) [inline AzureMonitorAgent/HandlerManifest.json:23]: azuremonitoragentmgr cgroup may have any number of mdsd tenants running under it in 1p case. as such, we cannot reasonably set a limit

**Review by simathih — APPROVED** [review]

**Comment by hestolz** [comment]
> to clarify, 250 means upto 250% on a 4 core machine? curious because, 250% on 3 core vs 4 core would leave different amounts of room behind for other applications

250 means 2.5 cores. This setting is directly translated by guest agent into systemd `CPUQuota` attribute for the service cgroup https://www.freedesktop.org/software/systemd/man/latest/systemd.resource-control.html#CPUQuota=

**Comment by simathih** [comment]
> > to clarify, 250 means upto 250% on a 4 core machine? curious because, 250% on 3 core vs 4 core would leave different amounts of room behind for other applications
> 
> 250 means 2.5 cores. This setting is directly translated by guest agent into systemd `CPUQuota` attribute for the service cgroup https://www.freedesktop.org/software/systemd/man/latest/systemd.resource-control.html#CPUQuota=

is the higher value safe in that case customer has fewer cores? I am assuming there is a way for customer to override this and we have documents on how they can do it

**Comment by hestolz** [comment]
> > > to clarify, 250 means upto 250% on a 4 core machine? curious because, 250% on 3 core vs 4 core would leave different amounts of room behind for other applications
> > 
> > 
> > 250 means 2.5 cores. This setting is directly translated by guest agent into systemd `CPUQuota` attribute for the service cgroup https://www.freedesktop.org/software/systemd/man/latest/systemd.resource-control.html#CPUQuota=
> 
> is the higher value safe in that case customer has fewer cores? I am assuming there is a way for customer to override this and we have documents on how they can do it

Currently we have no limits. So this is better than what we have currently. We can't make this variable on our side depending on how many cores the machine has etc; in order to not impact high EPS customers, we need to avoid setting a cap that will interfere with their workload. In case override is desired, it can be set as documented [here](https://github.com/Azure/azure-vmextension-publishing/wiki/2.0-Partner-Guide-Handler-Design-Details#2711-override-extension-cpu-memory-limits-in-linux)

**Comment by simathih** [comment]
> > > > to clarify, 250 means upto 250% on a 4 core machine? curious because, 250% on 3 core vs 4 core would leave different amounts of room behind for other applications
> > > 
> > > 
> > > 250 means 2.5 cores. This setting is directly translated by guest agent into systemd `CPUQuota` attribute for the service cgroup https://www.freedesktop.org/software/systemd/man/latest/systemd.resource-control.html#CPUQuota=
> > 
> > 
> > is the higher value safe in that case customer has fewer cores? I am assuming there is a way for customer to override this and we have documents on how they can do it
> 
> Currently we have no limits. So this is better than what we have currently. We can't make this variable on our side depending on how many cores the machine has etc; in order to not impact high EPS customers, we need to avoid setting a cap that will interfere with their workload. In case override is desired, it can be set as documented [here](https://github.com/Azure/azure-vmextension-publishing/wiki/2.0-Partner-Guide-Handler-Design-Details#2711-override-extension-cpu-memory-limits-in-linux)

thanks! makes sense

**Review by simathih — APPROVED** [review]

**Review by KidiIT — COMMENTED** [review]
All reviewed and approved❤️🫶

**Comment by KidiIT** [comment]
All checked and approved❤️🫶

**Pull request merged**